src/legacy/core_plugins/data/public/shim/legacy_module.ts

```diff
@@ -27,50 +27,95 @@ import { npStart } from 'ui/new_platform';
 import { FilterBar } from '../filter';
 import { IndexPatterns } from '../index_patterns/index_patterns';
 
+/** @internal */
+export const createFilterBarDirective = () => {
+  return {
+    restrict: 'E',
+    template: '',
+    compile: (elem: any) => {
+      const child = document.createElement('filter-bar-helper');
+
+      // Copy attributes to the child directive
+      for (const attr of elem[0].attributes) {
+        child.setAttribute(attr.name, attr.value);
+      }
+
+      child.setAttribute('ui-settings', 'uiSettings');
+      child.setAttribute('doc-links', 'docLinks');
+      child.setAttribute('plugin-data-start', 'pluginDataStart');
+
+      // Append helper directive
+      elem.append(child);
+
+      const linkFn = ($scope: any) => {
+        $scope.uiSettings = npStart.core.uiSettings;
+        $scope.docLinks = npStart.core.docLinks;
+        $scope.pluginDataStart = npStart.plugins.data;
+      };
+
+      return linkFn;
+    },
+  };
+};
+
+/** @internal */
+export const createFilterBarHelper = (reactDirective: any) => {
+  return reactDirective(wrapInI18nContext(FilterBar), [
+    ['uiSettings', { watchDepth: 'reference' }],
+    ['docLinks', { watchDepth: 'reference' }],
+    ['onFiltersUpdated', { watchDepth: 'reference' }],
+    ['indexPatterns', { watchDepth: 'collection' }],
+    ['filters', { watchDepth: 'collection' }],
+    ['className', { watchDepth: 'reference' }],
+    ['pluginDataStart', { watchDepth: 'reference' }],
+  ]);
+};
+
+/** @internal */
+export const createApplyFiltersPopoverDirective = () => {
```

Comment:
If you follow the suggestion from above, replacing filter-bar usage with `kbn-top-nav` - all of this can be deleted.